titus-common-runtime/src/main/java/com/netflix/titus/runtime/connector/kubernetes/KubeApiClients.java

```diff
@@ -76,6 +88,24 @@ public static ApiClient createApiClient(String kubeApiServerUrl,
         return client;
     }
 
+    public static Either<Boolean, Throwable> checkKubeConnectivity(ApiClient apiClient) {
```

Comment:
these two methods (`checkKubeConnectivity` and `mustHaveKubeConnectivity`) are not being used anywhere here. Are they intended to be provided as a library?

The `boolean` return is also not being used anywhere (see comment below)

Reply:
Yes, they are used in nftitus.
The Either contract in this context makes it sufficient to only check if the error is present.

Reply:
I will change it to `Optional` to avoid this confusion.